app/controllers/publish_document_controller.rb

```diff
@@ -4,7 +4,7 @@ class PublishDocumentController < ApplicationController
   def confirmation
     @document = Document.find_by_param(params[:id])
 
-    if PublishingRequirements.new(@document).errors?(tried_to_publish: true)
+    if Requirements::Checker.new(@document).pre_publish_issues(raise_exceptions: true).any?
```

Comment:
I found the raise_exceptions argument a bit confusing as I read it as I expected that to mean we wanted to raise exceptions for issues rather than return an array (I was initially going to comment asking if `any?` did anything)

If I understand right this raise_exceptions is for when we fail to check if they added topics or not - working on that assumption I don't think we actually want to raise an exception and instead just have an issue that they need to require topics? so argument could be like `require_topics_check` or similar - what do you think?

Reply:
It is actually intended that we raise an exception and fail - this is documented on the card. Without the Publishing API we can't know if the document meets the requirements or not, but we also don't want to show an incorrect message 'Add a topic' to the user. Telling a little lie and saying their publish operation 'failed' was decided to be the best behaviour. This is why the flag is generically 'raise_exceptions', as it applies for anything where the check is undecidable.

Reply:
Resolved as discussed ^^^.